### PR TITLE
fix(AnalyticalTable - TypeScript): correct typing for `onRowClick` and accessor function

### LINKED ISSUE #6519: [AnalyticalTable]: Typing definition is incorrect for `onRowClick`
### Describe the bug

The type of `onRowClick` of interface `AnalyticalTablePropTypes` is defined in file `packages/main/src/components/AnalyticalTable/types/index.ts`:
```Typescript
  onRowClick?: (e: OnRowClickEvent) => void;
```
Which is:
```Typescript
interface OnRowClickEvent extends Omit<MouseEvent, 'detail'> {
  detail: { detail: { row: unknown; nativeDetail: number } };
}
```
This is incorrect, based on the code in `packages/main/src/components/AnalyticalTable/hooks/useSingleRowStateSelection.ts`, it should be one level of `detail`, not nested two levels.

### Isolated Example

_No response_

### Reproduction steps

N/A

### Expected Behaviour

Better typing definition:
```Typescript
interface OnRowClickEvent extends Omit<UIEvent, 'detail'> {
  detail: { row: unknown; nativeDetail: number };
}
```

### Screenshots or Videos

_No response_

### UI5 Web Components for React Version

2.3.1

### UI5 Web Components Version

2.3.0

### Browser

Chrome, Edge, Firefox, Safari

### Operating System

_No response_

### Additional Context

_No response_

### Relevant log output

_No response_

### Organization

_No response_

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

### PR DESCRIPTION
**Pull request by Jianrong-Yu**
Fix #6519

Based on the code in `packages/main/src/components/AnalyticalTable/hooks/useSingleRowStateSelection.ts`, the type of `OnRowClickEvent` should be 
```Typescript
{
  detail: { row: unknown; nativeDetail: number };
}
```
And the type `MouseEvent` should be changed to `UIEvent` since the event can also be a Keyboard event based on the code:
```Typescript
  const handleKeyDown = (e) => {
    if ((!e.target.hasAttribute('aria-expanded') || (e.shiftKey && e.code === 'Space')) && e.code === 'Enter') {
      if (
        !webComponentsReactProperties.tagNamesWhichShouldNotSelectARow.has(
          getTagNameWithoutScopingSuffix(e.target.tagName)
        )
      ) {
        e.preventDefault();
      }
      handleRowSelect(e);
    }
    if (e.code === 'Space') {
      e.preventDefault();
    }
  };
```

### CODE REVIEW

File: packages/main/src/components/AnalyticalTable/types/index.ts [hunk 1]
```diff
@@ -329,7 +329,7 @@ export interface AnalyticalTableColumnDefinition {
    * __Note__: You can also specify deeply nested values with accessors like `info.hobby` or even `address[0].street`
    * __Note__: If no `accessor` is set, or the `accessor` is a function, the `id` property has to be set.
    */
-  accessor?: string | ((row: RowType, rowIndex: number) => any);
+  accessor?: string | ((row: Record<string, any>, rowIndex: number) => any);
```
Comment (Lukas742) [inline packages/main/src/components/AnalyticalTable/types/index.ts:332]: It looks like we not only used the wrong type here, but also missed typing some additional parameters:

```suggestion
  accessor?: string | ((originalRow: Record<string, any>, rowIndex: number, row: RowType, allRows: RowType[], data: Record<string, any>[]) => any);
```

File: packages/main/src/components/AnalyticalTable/types/index.ts [hunk 2]
```diff
@@ -541,8 +541,8 @@ interface OnAutoResizeMouseEvent extends Omit<MouseEvent, 'detail'> {
   detail: { columnId: string; width: number; nativeDetail: 2 };
 }
 
-interface OnRowClickEvent extends Omit<MouseEvent, 'detail'> {
-  detail: { detail: { row: unknown; nativeDetail: number } };
+interface OnRowClickEvent extends Omit<UIEvent, 'detail'> {
+  detail: { row: unknown; nativeDetail: number };
```
Comment (Lukas742) [inline packages/main/src/components/AnalyticalTable/types/index.ts:545]: `RowType` can be used here as well:

```suggestion
  detail: { row: RowType; nativeDetail: number };
```

**Review by Lukas742 — CHANGES_REQUESTED** [review]
Hi @Jianrong-Yu thanks a lot for your contribution!

**Comment by Jianrong-Yu** [comment]
@Lukas742 Thanks for you suggestions, the code is modified now.

File: packages/main/src/components/AnalyticalTable/types/index.ts [hunk 3]
```diff
@@ -329,7 +329,7 @@ export interface AnalyticalTableColumnDefinition {
    * __Note__: You can also specify deeply nested values with accessors like `info.hobby` or even `address[0].street`
    * __Note__: If no `accessor` is set, or the `accessor` is a function, the `id` property has to be set.
    */
-  accessor?: string | ((row: RowType, rowIndex: number) => any);
+  accessor?: string | ((originalRow: Record<string, any>, rowIndex: number, row: RowType, parentRows: RowType[], data: Record<string, any>[]) => any);
```
Comment (Lukas742) [inline packages/main/src/components/AnalyticalTable/types/index.ts:332]: Please run `prettier` to format the code.

```sh
prettier <file-path> --write
```

**Review by Lukas742 — CHANGES_REQUESTED** [review]

File: packages/main/src/components/AnalyticalTable/types/index.ts (hunk 3 above)
Comment (Jianrong-Yu) [reply packages/main/src/components/AnalyticalTable/types/index.ts]: Done.

**Review by Lukas742 — APPROVED** [review]
LGTM :ship:

Thanks again for your contribution!

**Pull request merged**